### PR TITLE
Adjust ARM's TARGET_HW_MACHINE_ARCH based on __ARM_ARCH based on arch(7)

### PR DESCRIPTION
**Pull request by kevans91**
In theory this addresses issue #53, and in theory it works...

### CODE REVIEW

**Comment by kevans91** [comment]
I think this warrants a little bit of testing. Also, I'm not sure of the logistics, but I think we probably need to figure out how to do an exp-run with this to establish what kind of fallout this might have on an armv7 jail.

**Comment by kevans91** [comment]
Argh, this isn't quite right... will fix later.

File: bsd-user/arm/target_syscall.h [hunk 1]
```diff
@@ -31,6 +31,12 @@ struct target_pt_regs {
 #define TARGET_FREEBSD_ARM_GET_TP       3
 
 #define TARGET_HW_MACHINE       "arm"
+#if __ARM_ARCH >= 7
+#define TARGET_HW_MACHINE_ARCH  "armv7"
+#elif __ARM_ARCH == 6
 #define TARGET_HW_MACHINE_ARCH  "armv6"
+#else
+#define TARGET_HW_MACHINE_ARCH  "arm"
```
Comment (bsdimp) [inline bsd-user/arm/target_syscall.h]: #error THIS DOESNT WORK

**Comment by bsdimp** [comment]
So how's it not quite right? Seems fine to me.

**Comment by kevans91** [comment]
We only build one flavor of qemu-*-static for all of 32-bit ARM so we're guaranteed to get it wrong for someone, somewhere.

**Comment by bsdimp** [comment]
Isn't there some way to know if you are emulating FreeBSD 11 or 12 or 13? You can key it based on that.

**Comment by kevans91** [comment]
@bsdimp I think this is updated to the version I'm going with... I tried to write get_target_arch to be generally useful in case I identify other things that need to care about this while I'm mucking around, though clearly it's only used for this at the moment.

We could perhaps also use it to identify a default -cpu to emulate, though I don't know if there's any point.

**Review by bsdimp — APPROVED** [review]
Looks sane to me..

**Pull request merged**